pkg/kubelet/kubelet.go

```diff
@@ -2353,6 +2353,18 @@ func (kl *Kubelet) generatePodStatus(pod *api.Pod) (api.PodStatus, error) {
 	podFullName := kubecontainer.GetPodFullName(pod)
 	glog.V(3).Infof("Generating status for %q", podFullName)
 
+	if existingStatus, ok := kl.statusManager.GetPodStatus(pod.UID); ok {
+		// This is a hacky fix to ensure container restart counts increment
+		// monotonically. Normally, we should not modify given pod. In this
+		// case, we check if there are cached status for this pod, and update
+		// the pod so that we update restart count appropriately.
+		// TODO(yujuhong): We will not need to count dead containers every time
+		// once we add the runtime pod cache.
+		// Note that kubelet restarts may still cause temporarily setback of
```

Comment:
Agreed! Labels are probably not the solution for all persistency problems.
Let's see how far we can get by with labels :) One more point - we will
need the existing logic, to handle backwards compat.

On Mon, Aug 24, 2015 at 2:21 PM, Yu-Ju Hong <notifications@github.com>
wrote:

> In pkg/kubelet/kubelet.go
> <https://github.com/kubernetes/kubernetes/pull/13103#discussion_r37806368>
> :
>
> > @@ -2353,6 +2353,18 @@ func (kl *Kubelet) generatePodStatus(pod *api.Pod) (api.PodStatus, error) {
> >  	podFullName := kubecontainer.GetPodFullName(pod)
> >  	glog.V(3).Infof("Generating status for %q", podFullName)
> >
> > +	if existingStatus, ok := kl.statusManager.GetPodStatus(pod.UID); ok {
> > +		// This is a hacky fix to ensure container restart counts increment
> > +		// monotonically. Normally, we should not modify given pod. In this
> > +		// case, we check if there are cached status for this pod, and update
> > +		// the pod so that we update restart count appropriately.
> > +		// TODO(yujuhong): We will not need to count dead containers every time
> > +		// once we add the runtime pod cache.
> > +		// Note that kubelet restarts may still cause temporarily setback of
>
> The thing is...kubelet has other persistence issues, and this is not a
> general solution :) We'd need to develop *some* solution eventually. I
> will update the PR later with the label change.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/13103/files#r37806368>.
>